crates/client/rpc/src/lib.rs

```diff
@@ -1714,43 +1872,60 @@ where
     })
 }
 
-async fn convert_tx_to_extrinsic<C, B>(
-    client: Arc<C>,
-    best_block_hash: <B as BlockT>::Hash,
-    transaction: UserTransaction,
-) -> Result<<B as BlockT>::Extrinsic, StarknetRpcApiError>
-where
-    B: BlockT,
-    C: ProvideRuntimeApi<B>,
-    C::Api: StarknetRuntimeApi<B> + ConvertTransactionRuntimeApi<B>,
-{
-    let extrinsic = client.runtime_api().convert_transaction(best_block_hash, transaction).map_err(|e| {
-        error!("Failed to convert transaction: {:?}", e);
-        StarknetRpcApiError::InternalServerError
-    })?;
+fn h256_to_felt(h256: H256) -> Result<FieldElement, Felt252WrapperError> {
+    Felt252Wrapper::try_from(h256).map(|f| f.0)
+}
 
-    Ok(extrinsic)
+/// The current timestamp in seconds.
+fn calculate_pending_block_timestamp() -> u64 {
+    let timestamp_in_millisecond = sp_timestamp::InherentDataProvider::from_system_time().as_millis();
+    timestamp_in_millisecond / 1000
 }
 
-fn convert_error<C, B, T>(
-    client: Arc<C>,
-    best_block_hash: <B as BlockT>::Hash,
-    call_result: Result<T, DispatchError>,
-) -> Result<T, StarknetRpcApiError>
-where
-    B: BlockT,
-    C: ProvideRuntimeApi<B>,
-    C::Api: StarknetRuntimeApi<B> + ConvertTransactionRuntimeApi<B>,
-{
-    match call_result {
-        Ok(val) => Ok(val),
-        Err(e) => match client.runtime_api().convert_error(best_block_hash, e) {
-            Ok(starknet_error) => Err(starknet_error.into()),
-            Err(_) => Err(StarknetRpcApiError::InternalServerError),
-        },
+fn is_pending_block(block_id: BlockId) -> bool {
+    block_id == BlockId::Tag(BlockTag::Pending)
+}
+
+fn calculate_l1_message_hash(_tx: &HandleL1MessageTransaction) -> Hash256 {
+    // TODO(#1291): compute message hash correctly to L1HandlerTransactionReceipt
+    Hash256::from_felt(&FieldElement::default())
```

Comment:
we have `impl ComputeTransactionHash for HandleL1MessageTransaction` in   `crates/primitives/transactions/src/compute_hash.rs`. Why not use it?